Realm/RLMRealm.mm

```diff
@@ -780,4 +787,158 @@ - (void)detachAllEnumerators {
     _collectionEnumerators = nil;
 }
 
+#pragma mark - Async Writes
+
+- (void)writeAsyncWithBlock:(RLMAsyncWriteBlock)block completion:(RLMCompletionBlock)completion {
+    RLMRealmConfiguration *config = self.configuration;
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        NSError *error = nil;
+        RLMRealm *bgRealm = [RLMRealm realmWithConfiguration:config error:&error];
+        
+        if (error) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                completion(error);
+            });
+            return;
+        }
+        
+        [bgRealm transactionWithBlock:^{
+            block(bgRealm);
+        } error:&error];
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
adjusted to fire the completion block off the notification, let me know if it looks right